### ISSUE TITLE
Injecting ff-record-list via XHR does not render ff-record

### ISSUE THREAD

**Issue by skurrilo** [FACT-Finder-Web-Components/ff-web-components#23]
i'm developing a javascript, which can be included in three different systems. for this reason and i18n i'm loading the template for the suggest and record-list via an XHR. Suggest template works fine, but the record-lists gets only empty ff-record tags after a search is triggered.

this is small example, based on the ff-record-list example:

```HTML
<!DOCTYPE html>
<html lang="en" class="home no-js">
<head>
    <meta charset="utf-8">
    <title>Record List</title>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js" type="text/javascript"></script>
    <script src="ff/ff-web-components/dist/vendor/custom-elements-es5-adapter.js"></script>
    <script src="ff/ff-web-components/dist/vendor/webcomponents-loader.js"></script>
    <script defer src="ff/ff-web-components/dist/bundle.js"></script>
</head>
<body>
<ff-communication url="https://search-web-components.fact-finder.de/FACT-Finder-7.3"
                  version="7.3"
                  default-query="bagpack"
                  channel="bergfreunde-co-uk">
</ff-communication>

<ff-searchbox>
    <input />
</ff-searchbox>

<ff-searchbutton><button>Search</button></ff-searchbutton>

<div class="container" id="factfinder_result">
    <!-- placeholder for rendering the search result template -->
</div>
<script type="text/javascript">
    let searchResultContainer = $('#factfinder_result');
    let template = '{{=<% %>=}}<ff-record-list unresolved>\n' +
        '    <ff-record>\n' +
        '        <img data-image="{{record.ImageName}}">\n' +
        '        <a data-anchor="https://www.alpinetrek.co.uk{{record.Deeplink}}"\n' +
        '           data-redirect="https://www.alpinetrek.co.uk{{record.Deeplink}}" data-redirect-target="_blank">\n' +
        '            {{record.Title}}\n' +
        '        </a>\n' +
        '\n' +
        '        <div class="priceContainer"><span class="brand">{{record.Manufacturer}}</span> <span class="price">{{record.Price}} €</span>\n' +
        '        </div>\n' +
        '    </ff-record>\n' +
        '</ff-record-list>\n';

    document.addEventListener('ffReady',function () {
        jQuery.ajax({
            url: document.location.origin + document.location.pathname, // just load this document again via ajax for timing purpose
            async: true,
            success: function (garbage) { // template already defined
                searchResultContainer.html(Mustache.render(template, {}));
            },
        });
    });
</script>
</body>
</html>
```
if i inject the template directly instead in the success method of the ajax-request, the ff-record gets rendered.

i tried to track it down and got as far as, that the template for the ff-record rendering is empty in the Mustache.render function. 

is there a know way to inject the ff-record-list template via javascript and XHR? I need to parse the template before adding it to the DOM for i18n.

i'm using version 3.5.0

**Comment by ghost** [comment]
This is indeed surprising tricky, due to some internal legacy lifecycle. We have already plans to clean things up there. 

Here is how I made it work:

```html
<!DOCTYPE html>
<html lang="en" class="home no-js">
<head>
    <meta charset="utf-8">
    <title>Record List</title>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js" type="text/javascript"></script>
    <script src="ff/ff-web-components/dist/vendor/custom-elements-es5-adapter.js"></script>
    <script src="ff/ff-web-components/dist/vendor/webcomponents-loader.js"></script>
    <script defer src="ff/ff-web-components/dist/bundle.js"></script>
</head>
<body>
<ff-communication url="https://search-web-components.fact-finder.de/FACT-Finder-7.3"
                  version="7.3"
                  default-query="bagpack"
                  channel="bergfreunde-co-uk">
</ff-communication>

<ff-searchbox>
    <input />
</ff-searchbox>

<ff-searchbutton><button>Search</button></ff-searchbutton>

<div class="container" id="factfinder_result">
    <ff-record-list unresolved>
        <ff-record></ff-record>
    </ff-record-list>
</div>
<script type="text/javascript">

    document.addEventListener('ffReady',function () {
        jQuery.ajax({
            url: document.location.origin + document.location.pathname, // just load this document again via ajax for timing purpose
            async: true,
            success: function (garbage) { // template already defined
                let searchResultContainer = $('#factfinder_result');
                let template =
                    '        <img data-image="{{record.ImageName}}">\n' +
                    '        <a data-anchor="https://www.alpinetrek.co.uk{{record.Deeplink}}"\n' +
                    '           data-redirect="https://www.alpinetrek.co.uk{{record.Deeplink}}" data-redirect-target="_blank">\n' +
                    '            {{record.Title}}\n' +
                    '        </a>\n' +
                    '\n' +
                    '        <div class="priceContainer"><span class="brand">{{record.Manufacturer}}</span> <span class="price">{{record.Price}} €</span>\n' +
                    '        </div>\n'
                document.querySelector('ff-record-list').insertElementHelper.elementTemplate.templateString = template;
            },
        });
    });
</script>
</body>
</html>
```

**Comment by ghost** [comment]
(As reference: For internal discussion I have created the internal feature request FFWEB-1323, which should elaborate the general advantage of dynamic templating)
